File: Reference Bots/C#/ReferenceBot/Domain/State/Cell.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Newtonsoft.Json;
using ReferenceBot.Domain.Command;

namespace ReferenceBot.Domain.State
{
    public class Cell
    {
        [JsonProperty]
        public bool Occupied { get; set; }
        
        [JsonProperty]
        public bool Hit { get; set; }

        [JsonIgnore]
        public Point Point => new Point(X, Y);

        [JsonProperty]
        public int X { get; set; }

        [JsonProperty]
        public int Y { get; set; }
    }
}

File: Sample Bots/Java/SampleBot/src/main/java/za/co/entelect/challenge/domain/state/Cell.java
package za.co.entelect.challenge.domain.state;


import za.co.entelect.challenge.domain.command.Point;

public class Cell {

    public boolean Occupied;

    public boolean Hit;

    public int X;

    public int Y;

    public Point getPoint() {

        return new Point(X, Y);
    }
}
